Sources/SwiftDocC/Infrastructure/Link Resolution/PathHierarchy.swift

```diff
@@ -352,7 +352,7 @@ struct PathHierarchy {
         newNode.identifier = newReference
         self.lookup[newReference] = newNode
         
-        modules[name] = newNode
+        modules.append(newNode)
```

Comment:
Hi @d-ronnqvist. It seems that this is what fixes the bug, and the rest is refactoring as a consequence of changing modules from a dictionary to an array.

However, I'm confused about some other changes that don't appear to have any impact on resolving the bug. I'll add a comment to each of the things I believe are unnecessary in this PR, so you can explain me why these changes were made.